mygit.rb

```diff
@@ -0,0 +1,365 @@
+        puts "<MyGit> : Git Credentials have been successfully configured"
+end
+
+
+def octoKitInit()
+        $LOG.debug("<MyGit> <octoKitInit>  : Initializing Octokit API for git")
+        gitconfig = ReturnGitCredentials()
+        login = gitconfig[0]
+        password = gitconfig[1]
+        client = Octokit::Client.new(:login => login , :password => password)
+        $LOG.debug("<MyGit> <octoKitInit> : Successfully logged into Octokit")
+        return client
+end
+
+def GitExist?
+        $LOG.debug("<MyGit> : <GitExist?> : Checking if the user has git installed")
+        git_path = `which git`
+        if git_path.length > 0
+                $LOG.debug("<MyGit> : <GitExist?> : Git exists")
+                return true
+
+        else
+                $LOG.debug("<MyGit> : <GitExist?> : Git isnt installed, requesting the user to install git")
+                puts "<MyGit> : <Error> : git isn`t installed on your system, kindly install and run the app again!"
+                return false
+        end
+        
+end     
+
+
+def getreponame
+        $LOG.debug("<MyGit> : <getreponame> : Asking for reponame")
+        repo = File.open('repo.txt', 'r').read
+        repo = repo.split('/')[1]
+        return repo
+
+end
+
+def mygit_create(repoandbranch)
+        $LOG.debug("<MyGit> <mygit_create> : mygit_create called with repo: #{repoandbranch[1]} and branch: #{repoandbranch[2]}")
+        $LOG.debug("<MyGit> <mygit_create> : Checking for git exists : #{GitExist?()}")
+        if !GitExist?()
+                $LOG.debug("<MyGit> <mygit_create> : Exiting because git doesn`t exist")
+                exit
+        end
+        repo = repoandbranch[1]
+        branch =  repoandbranch[2]
+        $LOG.debug("<MyGit> <mygit_create> : Starting to clone the repo #{repo}")
+        system "git clone #{repo}"
+        puts "<MyGit> : Successfully Cloned #{repo}"        
+        $LOG.debug("<MyGit> <mygit_create> : Successfully cloned #{repo}")
+        #Get the last modified folder (which will be the cloned repo) so that
+        #we can cd into it
+       
+        lastmod = getlastmodified()
+        $LOG.debug("<MyGit> <mygit_create> : last modified directory : #{lastmod}")
+       
+        #cd into it
+       
+        pw = `pwd`
+        pw=pw.chomp
+                                        
+        Dir.chdir "#{lastmod}"
+        username = repo.split('/')[3]
+        repo=(repo.split('/')[4]).split('.')[0]
+        puts repo
+        repolink  = "#{username}"+"/"+"#{lastmod}"
+        Dir.mkdir(lastmod)
+        `mv #{lastmod} .#{lastmod}`
+        $LOG.debug("<MyGit> <mygit_create> : Created the Hidden Directory to store data")
+        system "git init"
+        system "git pull"
+
+        if !(branch==repo)
+                system "git branch #{branch}"
+                system "git checkout #{branch}"
+                writetofile("Direx.txt",branch,repo)
```

Comment:
File names like repo.txt / Direx.txt / gitconfig.txt , must be constants , If the filename needs to be changed then only the constants value needs to be edited.